src/test/java/teammates/test/pageobjects/AppPage.java

```diff
@@ -901,12 +897,8 @@ private static String replaceUnpredictableValuesWithPlaceholders(String content)
                 // today's date
                 .replace(TimeHelper.formatDate(now).replace("/", "&#x2f;"), "${today}")
                 .replace(TimeHelper.formatDate(now), "${today}")
-                // now (used in comments last edited date) e.g. [Thu, 07 May 2015, 07:52:13 UTC]
-                .replaceAll(new SimpleDateFormat("EEE, dd MMM yyyy, ").format(now) + "[0-9]{2}:[0-9]{2}:[0-9]{2} UTC", "\\${comment\\.date}")
-                // now (date, time)
-                .replaceAll(new SimpleDateFormat("EEE, dd MMM yyyy, ").format(now) + "[0-9]{2}:[0-9]{2} [AP]M", "\\${datetime\\.now}")
-                // dynamic feedback submission numbers
-                .replaceAll("(?s)<span class=\"submissionsNumber\".*?</span>", "<span class=\"submissionsNumber\" id=\"submissionsNumber\">\\${submissions\\.number}</span>")
+                // now (date, time) e.g. [Thu, 07 May 2015, 07:52 PM] or [Thu, 07 May 2015, 07:52:13 UTC]
+                .replaceAll(new SimpleDateFormat("EEE, dd MMM yyyy, ").format(now) + "[0-9]{2}:[0-9]{2}( [AP]M|:[0-9]{2} UTC)", "\\${datetime\\.now}")
```

Comment:
No, that is wrong. It might be true for certain tests, but as long as the full test suite is concerned, _all_ is covered. This is what I've been trying to say from the beginning: we have hard-coded test cases that will cover such cases to prevent flexibility even in things as small as display formats.
By using same placeholder, yes I am weakening the test _which uses the test accounts or date/time now_, but there are hundreds of hard-coded test cases that cover it and the weakening becomes a very small price to pay.

This calls for concrete examples:
1. [This file](https://github.com/TEAMMATES/repo/blob/master/src/test/resources/pages/InstructorCourseStudentDetailsRegistered.html). The google ID in question is `CCSDetailsUiT.instr.veryLongGoogleId`, long enough to necessitate truncation. And in that test file, _it is very specific_ where `CCSDetailsUiT.instr.veryLongGoogleId` and `CCSDetailsUiT.instr....` have to appear. Had someone change where truncation occurs without changing this test file, dev green is a big no.
2. [This file](https://github.com/TEAMMATES/repo/blob/master/src/test/resources/pages/studentCourseDetailsWithTeammatesHTML.html). Same case as above, the google ID is `SCDetailsUiT.alice.veryLongGoogleId`.
3. Now for the comment date/time format: [see how much changes are made in the HTML files just by changing the date/time format](https://github.com/TEAMMATES/repo/compare/4585-reduce-placeholders...4590-datetime-comments?w=1). The message is the same: _it is very specific_ where the date/time format for comments appear, and we don't need another placeholder just to make sure of that.